### PR TITLE
Remove the last uses of the legacy refiner from Equality.

### PR DESCRIPTION
**Pull request by ppedrot**
Together with #16171 this finishes the porting of the Equality module to the new unification engine.

### CODE REVIEW

**Comment by ppedrot** [comment]
@coqbot bench

**Review by Alizter — APPROVED** [review]
Looks pretty good. Bench is looking good too. I wonder if the check_flag argument could be made more clear. `bool option` is not very readable IMO, but that's ok if it is done in another PR.

File: tactics/equality.ml [hunk 1]
```diff
@@ -1520,29 +1522,30 @@ let decompEqThen keep_proofs ntac eq =
...
-      match find_positions env sigma ~keep_proofs ~no_discr:false t1 t2 with
-      | Inl (cpath, (_,dirn), _) ->
-          discr_positions env sigma eq cpath dirn
-      | Inr [] -> (* Change: do not fail, simplify clear this trivial hyp *)
-        ntac eq.eq_term 0
+    let ido = try Some (destVar sigma eq.eq_term) with DestKO -> None in
+    match find_positions env sigma ~keep_proofs ~no_discr:false t1 t2 with
+    | Inl (cpath, (_,dirn), _) ->
+      discr_positions env sigma eq cpath dirn
+    | Inr [] -> (* Change: do not fail, simplify clear this trivial hyp *)
```
[the remark this answers is not in the document]
Comment (ppedrot) [reply tactics/equality.ml:1529]: I think that this is about equalities on unit types, they can't be injected further. The code just removes such equations, if the empty case was not handled specifically it wouldn't work.

**Comment by Alizter** [comment]
I'll give this time to sit and attract reviewers and think about merging it next week.

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**